### PR TITLE
Removing unnecessary lambdas part 38

### PR DESCRIPTION
**Pull request by Piotrek321**

### CODE REVIEW

File: src/blockchain/node/HeaderOracle.cpp [hunk 1]
```diff
@@ -438,18 +431,17 @@ auto HeaderOracle::best_hashes(
     const auto limitIsZero = (0 == limit);
     auto current{start};
     const auto tip = best_chain(lock);
-    const auto last = [&] {
-        if (limitIsZero) {
+    block::Height last{};
```
Comment (wiesiekpap) [inline src/blockchain/node/HeaderOracle.cpp]: I would speculatively initialize it to tip.first and change it if(!limitZero), without an else clause.

File: src/blockchain/node/HeaderOracle.cpp [hunk 2]
```diff
@@ -963,34 +935,28 @@ auto HeaderOracle::ProcessSyncData(
...
+        block::Hash previous{};
+        const auto height = blocks.front().Height();
 
-                return block::Hash{};
-            } else {
-                const auto rc = prior.Assign(database_.BestBlock(height - 1));
-
-                OT_ASSERT(rc);
+        if (0 >= height) {
+            previous = block::Hash{};
```
Comment (wiesiekpap) [inline src/blockchain/node/HeaderOracle.cpp]: This assignment is redundant.

**Review by wiesiekpap — APPROVED** [review]
A minor issue of a redundant assignment should be fixed before someone finds a real static analysis tool that finds this for us.

File: src/blockchain/node/HeaderOracle.cpp (hunk 2 above)
Comment (Piotrek321) [reply src/blockchain/node/HeaderOracle.cpp]: yeah... :D

**Pull request merged**